library/core/src/slice/ascii.rs

```diff
@@ -174,6 +176,9 @@ impl_fn_for_zst! {
 #[must_use = "iterators are lazy and do nothing unless consumed"]
 pub struct EscapeAscii<'a> {
     inner: iter::FlatMap<super::Iter<'a, u8>, ascii::EscapeDefault, EscapeByte>,
+
+    /// Cached length of the iterator to implement `ExactSizeIterator`.
+    len: usize,
```

Comment:
you can't cache the value because len() must be updated every time an item is consumed from the iterator.

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.size_hint

> Returns the bounds on the _remaining_ length of the iterator.

Reply:
Thank you for pointing out my oversight. I will recommit.

Reply:
well, caching may still be cheaper than recomputing it every time, but you'll at least have to update the cached value every time an item is consumed.

Reply:
I feel caching is the better choice even with the extra overhead of an decrement each `next()` or `next_back()` and an extra `usize` for each `EscapeAscii`. Caching avoids degenerate worst cases like users calling `len()` or `size_hint()` after each byte received from a slow source. 

It's unfortunately to even have to go through the source iterator more than once but there isn't any other way to implement `ExactSizeIterator` correctly and it can't be removed because it's been released in a stable version.

Reply:
> and it can't be removed because it's been released in a stable version.

The stability policy does have exceptions for accidental stabilizations and critical bugfixes. So removing it is an option. The crater run in #99880 will tell us whether it's worthwhile taking that option.